Tests/SwiftSyntaxTest/SymbolNameTests.swift

```diff
@@ -0,0 +1,210 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import XCTest
+
+final class SymbolNameTests: XCTestCase {
+
+  // MARK: - Functions
+
+  func testFunctionNameWithoutParameters() {
+    assertSymbolName(ofFunction: "func name() {}", expecting: "name()")
+  }
+
+  func testFunctionNameWithSingleNamedParameter() {
+    assertSymbolName(ofFunction: "func name(one: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithSingleUnnamedParameter() {
+    assertSymbolName(ofFunction: "func name(_: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(one two: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithExplicitExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(one _: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithImplicitExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(_ two: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithOnlyAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(_ _: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithMultipleNamedParameters() {
+    assertSymbolName(ofFunction: "func name(one two: Int, three: Int) {}", expecting: "name(one:three:)")
+  }
+
+  func testFunctionNameWithAllParametersHavingExternalNames() {
+    assertSymbolName(ofFunction: "func name(one two: Int, three four: Int) {}", expecting: "name(one:three:)")
+  }
+
+  func testFunctionNameWithMixedNamedAndAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(one two: Int, _ four: Int) {}", expecting: "name(one:_:)")
+  }
+
+  func testFunctionNameWithAllAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(_ two: Int, _ four: Int) {}", expecting: "name(_:_:)")
+  }
+
+  func testFunctionNameWithBackticks() {
+    assertSymbolName(ofFunction: "func `name`() {}", expecting: "name()")
+  }
+
+  func testFunctionNameWithParameterNameInBackticks() {
+    assertSymbolName(ofFunction: "func name(`one`: Int) {}", expecting: "name(one:)")
+  }
+
+  // FIXME: Does this make sense?
+  func testOperatorFunctionName() {
+    assertSymbolName(ofFunction: "func == (one: Int, two: Int)", expecting: "==(one:two:)")
+  }
```

Comment:
Should the operator function have a `symbolName`?

Reply:
Yeah I think so.

Reply:
Yes, and I think the test makes sense.